qiskit/primitives/backend_estimator.py

```diff
@@ -209,9 +216,27 @@ def _transpile(self):
                     perm_pattern = list(range(transpiled_circuit.num_qubits))
 
             # 2. transpile diff circuits
-            transpile_opts = copy.copy(self.transpile_options)
-            transpile_opts.update_options(initial_layout=perm_pattern)
-            diff_circuits = transpile(diff_circuits, self.backend, **transpile_opts.__dict__)
+            passmanager = PassManager([SetLayout(perm_pattern)])
```

Comment:
It might be better to separate this part (generating pass manager) as a utility function or a class method.